assets/app/scripts/controllers/deploymentConfig.js

```diff
@@ -83,45 +83,50 @@ angular.module('openshiftConsole')
           // Loading of the page that will create deploymentConfigDeploymentsInProgress structure, which will associate running deployment to his deploymentConfig.
           $scope.deploymentConfigDeploymentsInProgress = DeploymentsService.associateRunningDeploymentToDeploymentConfig(deploymentsByDeploymentConfig);
-        } else if (action === 'ADDED' || (action === 'MODIFIED' && ['New', 'Pending', 'Running'].indexOf($filter('deploymentStatus')(deployment)) > -1)) {
-          // When new deployment id instantiated/cloned, or in case of a retry, associate him to his deploymentConfig and add him into deploymentConfigDeploymentsInProgress structure.
-          $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] = $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] || {};
-          $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName] = deployment;
-        } else if (action === 'MODIFIED') {
-          // After the deployment ends remove him from the deploymentConfigDeploymentsInProgress structure.
-          if (!$filter('deploymentIsInProgress')(deployment)){
-            delete $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName];
+        } else if (DeploymentsService.deploymentBelongsToConfig(deployment, $routeParams.deploymentconfig)) {
+          var deploymentName = deployment.metadata.name;
+          var deploymentConfigName = $routeParams.deploymentconfig;
+          switch (action) {
+            case 'ADDED':
+              $scope.unfilteredDeployments[deploymentName] = deployment;
+              // When new deployment is instantiated/cloned, associate him to his deploymentConfig and add him into deploymentConfigDeploymentsInProgress structure.
+              $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] = $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] || {};
+              $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName] = deployment;
+              deployment.causes = $filter('deploymentCauses')(deployment);
+              break;
+            case 'MODIFIED':
+              $scope.unfilteredDeployments[deploymentName] = deployment;
+              // When deployment is retried, associate him to his deploymentConfig and add him into deploymentConfigDeploymentsInProgress structure.
+              if ($filter('deploymentIsInProgress')(deployment)){
+                $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] = $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] || {};
+                $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName] = deployment;                
+              } else { // After the deployment ends remove him from the deploymentConfigDeploymentsInProgress structure.
+                delete $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName];
```

Comment:
because im collapsing the added and modified block im going to add a check
just in case

On Thu, Nov 12, 2015 at 4:09 PM, Sam Padgett <notifications@github.com>
wrote:

> In assets/app/scripts/controllers/deploymentConfig.js
> <https://github.com/openshift/origin/pull/5862#discussion_r44714767>:
>
> > +          switch (action) {
> > +            case 'ADDED':
> > +              $scope.unfilteredDeployments[deploymentName] = deployment;
> > +              // When new deployment is instantiated/cloned, associate him to his deploymentConfig and add him into deploymentConfigDeploymentsInProgress structure.
> > +              $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] = $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] || {};
> > +              $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName] = deployment;
> > +              deployment.causes = $filter('deploymentCauses')(deployment);
> > +              break;
> > +            case 'MODIFIED':
> > +              $scope.unfilteredDeployments[deploymentName] = deployment;
> > +              // When deployment is retried, associate him to his deploymentConfig and add him into deploymentConfigDeploymentsInProgress structure.
> > +              if ($filter('deploymentIsInProgress')(deployment)){
> > +                $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] = $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName] || {};
> > +                $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName] = deployment;
> > +              } else { // After the deployment ends remove him from the deploymentConfigDeploymentsInProgress structure.
> > +                delete $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName][deploymentName];
>
> Can $scope.deploymentConfigDeploymentsInProgress[deploymentConfigName]
> ever be undefined?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5862/files#r44714767>.
>